backend/internal/service/association.go

```diff
@@ -0,0 +1,63 @@
+package service
+
+import (
+	"fmt"
+	"shift/internal/entity"
+
+	"github.com/sirupsen/logrus"
+)
+
+type AssociationService struct {
+	bucketDB      entity.BucketDB
+	associationDB entity.AssociationDB
+}
+
+func NewAssociationService(bucketDB entity.BucketDB, associationDB entity.AssociationDB) *AssociationService {
+	return &AssociationService{
+		bucketDB:      bucketDB,
+		associationDB: associationDB,
+	}
+}
+
+func (s *AssociationService) CreateAssociation(req *entity.CreateAssociationRequest) (*entity.CreateAssociationResponse, error) {
+	ass, err := s.createAssociation(req)
+
+	if err != nil {
+		return nil, fmt.Errorf("cannot create association: %s", err)
+	}
+
+	return ass, nil
+}
+
+func (s *AssociationService) createAssociation(req *entity.CreateAssociationRequest) (*entity.CreateAssociationResponse, error) {
+	association := new(entity.AssociationEntity)
+
+	if err := association.FromCreationRequest(req); err != nil {
+		return nil, fmt.Errorf("parsing request into association entity: %w", err)
+	}
+	logrus.Tracef("Parsed assoication entity: %+v", association)
+
+	association, err := s.associationDB.CreateAssociation(association)
```

Comment:
we need to store the association image on our bucket and create a new record on our db as well - something similar to the user creation.